azure/functions/extension/extension_scope.py

```diff
@@ -0,0 +1,21 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+from enum import Enum
+
+
+class ExtensionScope(Enum):
+    """There are two valid scopes of the worker extension framework.
+
+    APPLICATION:
+        It is injected in AppExtensionBase._scope. Any implementation of
+        AppExtensionBase will be applied into all triggers.
+
+    FUNCTION:
+        It is injected in FuncExtensionBase._scope. Any implementation of
+        FuncExtensionBase requires initialization in customer's function app
+        trigger.
+    """
+    UNKNOWN = 0
```

Comment:
This will be used when customer incorrectly using our MetaClass to build their own ExtensionBase. (e.g. InvalidExtensionBase). This will make sure our MetaClass will block this kind of extension registration. See unit test case [**test_extension_scope_not_set**](https://github.com/Azure/azure-functions-python-library/pull/81/files#diff-71e0210c7c8feef9a3bb3d0cac24fca2d10ba3f5675cc401a1f960ee982b322dR206)

Reply:
Please add this explanation as a comment in the pydoc itself.